### PR TITLE
Update tvl calculations for Amulet Protocol

### PR DESCRIPTION
**Pull request by yym-mtrlabs**
**NOTE**

1. The protocol is usually listed within 24 hours of merging the PR
2. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
3. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
4. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
5. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/amulet-protocol exports TVL: 
        
 
 
    solana                    358.02 k
    
    total                    358.02 k

**Comment by g1nt0ki** [comment]
@yym-mtrlabs I think it is better if we have two listings one for liquid staking: counts sol used for minting amSOL and another listing for underwriting insurance sol+amSOL in your vaults

Can you please create a new adapter maybe named `projects/amulet-sol-staking/index.js` ?

**Comment by yym-mtrlabs** [comment]
> @yym-mtrlabs I think it is better if we have two listings one for liquid staking: counts sol used for minting amSOL and another listing for underwriting insurance sol+amSOL in your vaults
> 
> Can you please create a new adapter maybe named `projects/amulet-sol-staking/index.js` ?

@g1nt0ki If we create a new adapter, does that mean there will be two records when we search for Amulet Protocol in DefiLlama? Or we need to use a new Name when submit PR?

**Comment by g1nt0ki** [comment]
there will be two records and a combined/parent record/listing: example: https://defillama.com/protocol/aave

**Comment by yym-mtrlabs** [comment]
> 



> there will be two records and a combined/parent record/listing: example: https://defillama.com/protocol/aave

While checking https://defillama.com/protocol/aave, they have several different names like AAVE V2, AAVE Arc. Where do we need to define this name in PR?

**Comment by g1nt0ki** [comment]
you can write the preferred name for the two products here and we will fix it in the server repo.

**Comment by yym-mtrlabs** [comment]
> you can write the preferred name for the two products here and we will fix it in the server repo.

Hi @g1nt0ki , AAVE v1, AAVE v2, AAVE v3 they are different protocols. For us, we only have one single protocol. 

May I check whether we can make it the similar way as https://defillama.com/protocol/pancakeswap?
We still keep all information in same page, but in Include in TVL (optional), we can enable or disable liquid staking part.

**Comment by g1nt0ki** [comment]
the staking in pancake is different from your, it is for $CAKE token/protocol's gov token. In your case, it is liquid staking of SOL

**Comment by g1nt0ki** [comment]
@yym-mtrlabs the reason I am pushing for this split is, you have two different kind of tvl, underwriting - considered as tvl by all, 'liquid staking' - not always considered a tvl, we have a toggle to enable/disable it from overall tvl

**Comment by yym-mtrlabs** [comment]
> @yym-mtrlabs the reason I am pushing for this split is, you have two different kind of tvl, underwriting - considered as tvl by all, 'liquid staking' - not always considered a tvl, we have a toggle to enable/disable it from overall tvl

@g1nt0ki Thanks for the advice, we will split into two adapters.

**Comment by llamatester** [comment]
The adapter at projects/amulet-protocol exports TVL: 
        
 
 
    solana                    345.32 k
    
    total                    345.32 k

**Comment by llamatester** [comment]
The adapter at projects/amulet-liquidity-staking exports TVL: 
        
 
 
    solana                    177.04 k
    
    total                    177.04 k

**Comment by yym-mtrlabs** [comment]
### New Adapter: projects/amulet-liquidity-staking/index.js

##### Name (to be shown on DefiLlama):
Amulet Protocol

##### Preferred Name (to be shown on DefiLlama):
Amulet Liquidity Staking

##### Twitter Link:
https://twitter.com/AmuletProtocol

##### List of audit links if any:
https://files.amulet.org/public/AmuletGlobalMTRLabs.pdf

##### Website Link:
https://amulet.org/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
https://files.amulet.org/public/twitter/Avatar_Circle.png

##### Current TVL:
177.04 k USD

##### Chain:
Solana

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)


##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)


##### Short Description (to be shown on DefiLlama):
Amulet's optimized staking algorithm enables users to earn PoS staking rewards on Solana without locking in their funds. Users can get instant liquidity for their staked $SOL and participate in DeFi with $amtSOL, Amulet's $SOL derivative.

##### Token address and ticker if any:

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Liquid Staking

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):


##### forkedFrom (Does your project originate from another project):


##### methodology (what is being counted as tvl, how is tvl being calculated):
Amulet enables users to earn PoS staking rewards on Solana by staking SOL and mint amtSOL. Hence, the amount of SOL staked are counted as our TVL for Amulet Liquidity Staking which the value are calculated based on the price get from Coingecko.

File: projects/amulet-protocol/index.js [hunk 1]
```diff
@@ -19,10 +19,22 @@ async function tvl() {
     const accountSolBalances = await Promise.all(accountList.map(async(x) => new BN((await connection.getAccountInfo(new PublicKey(x))).lamports)))
     const solAmount = [new BN(posSolVirtualAmount), new BN(liqSolAmount), new BN(totalActiveBalance), ...accountSolBalances]
 
-    const tvl = solAmount.reduce((acc,x) => acc.add(x), new BN(0)).div(new BN(1000000000)).toNumber()
+    const tvlSOL = solAmount.reduce((acc,x) => acc.add(x), new BN(0)).div(new BN(1000000000)).toNumber()
```
Comment (g1nt0ki) [inline projects/amulet-protocol/index.js:22]: can you remove this? it is already counted in projects/amulet-liquidity-staking

File: projects/amulet-protocol/index.js [hunk 2]
```diff
@@ -19,10 +19,22 @@ async function tvl() {
...
+
+    const stakingInstanceState = await connection.getAccountInfo(new PublicKey("HNhPNHkp3RobeJzepNzyVyewtAaoF3QCCvtBTxKJVnRX"));
+    const liqStakedAmtSOLAmount = stakingInstanceState.data.slice(203, 211);
```
Comment (g1nt0ki) [inline projects/amulet-protocol/index.js:25]: sorry, it is not very clear, what does this refer to?

Comment (yym-mtrlabs) [reply projects/amulet-protocol/index.js:25]: Hi this is amtSOL amount staked in our single-sided amtSOL-aUWT pool

File: projects/amulet-protocol/index.js (hunk 1 above)
Comment (yym-mtrlabs) [reply projects/amulet-protocol/index.js:22]: Sure, will update and re-submit

**Comment by llamatester** [comment]
Error while running adapter at projects/amulet-protocol: 
        
 
  
    
    
    ReferenceError: connection is not defined
    Truncated error stack:
    
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/amulet-protocol/index.js:6:34)
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:60:31)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:167:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:155:29
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:150:44

**Comment by llamatester** [comment]
The adapter at projects/amulet-protocol exports TVL: 
        
 
 
    solana                    168.86 k
    
    total                    168.86 k

**Comment by llamatester** [comment]
The adapter at projects/amulet-liquidity-staking exports TVL: 
        
 
 
    solana                    177.51 k
    
    total                    177.51 k

File: projects/amulet-protocol/index.js (hunk 1 above)
Comment (yym-mtrlabs) [reply projects/amulet-protocol/index.js]: Updated and re-submitted.

**Comment by g1nt0ki** [comment]
looks good, will update the listing soon

**Comment by yym-mtrlabs** [comment]
> looks good, will update the listing soon

Thanks a lot!

**Pull request merged**